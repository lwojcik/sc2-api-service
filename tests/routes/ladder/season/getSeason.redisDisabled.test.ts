import fastify from 'fastify';
import server from '../../../../src/index';
import getConfig from '../../../helper';

describe('/ladder/season/:regionId (Redis disabled)', () => {
  const fastifyServer = fastify() as any;
  const url = '/ladder/season/1';

  beforeAll(async () => {
    await fastifyServer.register(server, getConfig(false));
  });

  afterEach(() => {
    fastifyServer.close();
  });

  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url });
    expect(res.payload).toMatchSnapshot();
  });
});
