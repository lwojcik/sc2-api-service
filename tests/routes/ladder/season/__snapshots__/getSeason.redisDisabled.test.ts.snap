// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ladder/season/:regionId (Redis disabled) returns correct response 1`] = `"{\\"status\\":200,\\"data\\":{\\"seasonId\\":9999,\\"number\\":\\"\\",\\"year\\":\\"\\",\\"startDate\\":\\"\\",\\"endDate\\":\\"\\"}}"`;
